### PR TITLE
Avoid SPPM is always rebuilt

### PR DESCRIPTION
**Pull request by metalefty**
Fixes AlmaLinux/build-system#19.

-----

I haven't tested at all yet because I'm still not sure how to run the modified code. Comments are welcome.

### CODE REVIEW

**Comment by metalefty** [comment]
This is my very first PR on AlmaLinux Build System. Let me know if I misunderstand or miss something.

File: build_node/builders/base_rpm_builder.py [hunk 1]
```diff
@@ -235,8 +236,8 @@ def build_packages(self, src_dir, spec_file=None):
 
         Parameters
         ----------
-        src_dir : str
-            Path to the src-RPM sources.
+        src: str
```
Comment (anfimovdm) [inline build_node/builders/base_rpm_builder.py:239]: you need to rename argument too

File: build_node/builders/base_rpm_builder.py [hunk 2]
```diff
@@ -139,8 +139,9 @@ def build(self):
                 finally:
                     os.chdir(cwd)
             else:
-                source_srpm_dir = self.unpack_sources()
-                spec_file = self.locate_spec_file(source_srpm_dir)
+                source_srpm_dir = self.unpack_sources(True) # download only
+                spec_file = None
```
Comment (anfimovdm) [inline build_node/builders/base_rpm_builder.py:143]: I guess it's unnecessary to set spec_file to None because we already do it on 89 line

**Review by anfimovdm — CHANGES_REQUESTED** [review]

**Comment by anfimovdm** [comment]
other then that looks great, after your fixes I guess I can try to test your changes locally

File: build_node/builders/base_rpm_builder.py [hunk 3]
```diff
@@ -247,39 +248,53 @@ def build_packages(self, src_dir, spec_file=None):
...
-                                          srpm_artifacts=True)
-        srpm_path = srpm_build_result.srpm
-        self.logger.info('src-RPM %s was successfully built', srpm_path)
-        excluded, reason = self.is_build_excluded(srpm_path)
-        if excluded:
-            raise BuildExcluded(reason)
+        elif os.path.isfile(src) and src.endswith('src.rpm'): # if SRPM archive is given
+            srpm_path = src
+        else:
+            raise RuntimeError("Neither a directory or a src-RPM archive: {0}".format(str(src)))
```
Comment (javihernandez) [inline build_node/builders/base_rpm_builder.py:277]: Watch for long lines, i.e., this one. 
Ideally, no more than 80 chars.

**Review by javihernandez — COMMENTED** [review]
Apart the changes that @anfimovdm already request, LGTM

**Comment by metalefty** [comment]
I'm busy this week for making our 8.8 release and adopting recent Red Hat's announcement. Will be address your review comments later.

File: build_node/builders/base_rpm_builder.py (hunk 2 above)
Comment (metalefty) [reply build_node/builders/base_rpm_builder.py:143]: Indeed.

File: build_node/builders/base_rpm_builder.py (hunk 1 above)
Comment (metalefty) [reply build_node/builders/base_rpm_builder.py:238]: Renamed.

**Comment by metalefty** [comment]
Addressed review comments.

**Comment by anfimovdm** [comment]
After the tests I figured out that we need to rebuild SRPM to detect if build excluded for some architecture or not.

**Comment by metalefty** [comment]
We can't detect if architecture is excluded unless we build SRPMs because "excluded" is a result of build.
Then, what about checking the result of the binary build (SRPM->RPM), too?  Will it work?

Here's the concept code:
```diff
diff --git a/build_node/builders/base_rpm_builder.py b/build_node/builders/base_rpm_builder.py
index 6cc5a10..fc25b9d 100644
--- a/build_node/builders/base_rpm_builder.py
+++ b/build_node/builders/base_rpm_builder.py
@@ -200,48 +200,51 @@ class BaseRPMBuilder(BaseBuilder):
     def build_binaries(self, srpm_path, definitions=None):
         """
         Builds binary RPM packages, saves build artifacts
         to the artifacts directory.
 
         Parameters
         ----------
         srpm_path : str
             Path to SRPM
         definitions : dict, optional
             Dictionary with mock optional definitions
         """
         rpm_result_dir = os.path.join(self.task_dir, 'rpm_result')
         rpm_mock_config = self.generate_mock_config(self.config, self.task)
         rpm_build_result = None
         with self.mock_supervisor.environment(rpm_mock_config) as mock_env:
             try:
                 rpm_build_result = mock_env.rebuild(
                     srpm_path,
                     rpm_result_dir,
                     definitions=definitions,
                     timeout=self.build_timeout,
                 )
             except MockError as e:
+                excluded, reason = self.is_build_excluded(e)
+                if excluded:
+                    raise BuildExcluded(reason)
                 rpm_build_result = e
                 raise BuildError(f'RPM build failed: {str(e)}')
             finally:
                 if rpm_build_result:
                     self.save_build_artifacts(rpm_build_result)
 
     @measure_stage('build_packages')
     def build_packages(self, src, spec_file=None):
         """
         Builds src-RPM and binary RPM packages, saves build artifacts to the
         artifacts directory.
 
         Parameters
         ----------
         src: str
             Path to the src-RPM archive or unpacked sources.
         spec_file : str, optional
             Spec file path. It will be detected automatically if omitted.
         """
         mock_defines = self.task.platform.data.get('definitions')
         self.logger.info('starting src-RPM build')
         srpm_result_dir = os.path.join(self.task_dir, 'srpm_result')
         os.makedirs(srpm_result_dir)
         srpm_mock_config = self.generate_mock_config(self.config, self.task,

```

**Comment by metalefty** [comment]
Replaced by #59

**Pull request closed without merging**